engine/tox.ini

```diff
@@ -14,7 +14,15 @@ banned-modules =
 #
 # --dist no = temporarily disable xdist as it's leading to flaky tests :(
 # https://github.com/grafana/oncall-private/issues/2733
-addopts = --dist no --no-migrations --color=yes --showlocals
+
+# From pytest-socket docs (https://github.com/miketheman/pytest-socket):
+# A plugin to use with Pytest to disable or restrict socket calls during tests to ensure network calls are prevented
+# --disable-socket = tests should fail on any access to socket or libraries using socket with a SocketBlockedErro
+# --allow-hosts = allow connections to the given hostnames/IPs.
+#   - localhost = our tests on CI use localhost as the host to connect to databases running locally in docker container
+#   - oncall-dev-mariadb = if you're running things locally, with a MariaDB instance running, there's a good chance the hostname will be this
+#   - calendar.google.com = used for some tests that fetch a real iCal file from Google
+addopts = --dist no --no-migrations --color=yes --showlocals --disable-socket --allow-hosts=localhost,oncall-dev-mariadb,calendar.google.com
```

Comment:
Why do we need to allow `calendar.google.com`? Is it something we could avoid?

Reply:
good point 👍 right now we have three tests which fetch _real_ `.ical` files from `calendar.google.com` ([code search](https://github.com/search?q=repo:grafana/oncall%20calendar.google.com&type=code)).

I'll refactor those tests, and save those `.ical` files locally + mock out the responses from this network call to return that file content, then I think we can remove this 👍

Reply:
That would be great! :rocket:

Reply:
https://github.com/grafana/oncall/pull/5315/commits/977380a03c746fe2db3fc8ecfd5e445eddcb5a8b

Reply:
(turns out that the https://calendar.google.com/calendar/ical/amixr.io_37gttuakhrtr75ano72p69rt78%40group.calendar.google.com/private-1d00a680ba5be7426c3eb3ef1616e26d/basic.ics file we were referencing doesn't exist anymore anyways; was 404ing)